File: frontend/src/types/job-position.ts
// =====================================================
// 职位管理类型定义
// @author Azir
// =====================================================

// 职位状态（用户手动设置）
export type PositionStatus = 'draft' | 'applied' | 'interviewing' | 'offered' | 'rejected' | 'withdrawn'

// 职位列表项
export interface JobPositionListItem {
  id: string
  companyName: string
  title: string
  status: PositionStatus
  nextInterviewDate?: string
  nextInterviewRound?: string
  interviewCount: number
  latestInterviewDate?: string
  createdAt: string
  updatedAt: string
}

// 面试简要信息
export interface InterviewBrief {
  id: string
  status: string
  date?: string
  overallResult?: string
  // 新增字段
  source?: string           // 面试来源（real/mock）
  roundType?: string        // 轮次类型
  roundName?: string        // 自定义轮次名称
  interviewType?: string    // 面试类型（onsite/online）
  location?: string         // 现场地址
  onlineLink?: string       // 线上链接
}

// 职位详情
export interface JobPositionDetail {
  id: string
  companyId: string
  companyName: string
  title: string
  status: PositionStatus
  jdContent?: string
  jdAnalysis?: string
  jdAnalysisUpdatedAt?: string
  interviews: InterviewBrief[]
  createdAt: string
  updatedAt: string
}

// 创建职位请求
export interface CreateJobPositionRequest {
  companyName: string
  title: string
  jdContent?: string
}

// 更新职位请求
export interface UpdateJobPositionRequest {
  companyName?: string
  title?: string
  jdContent?: string
}

File: backend/src/main/java/com/careerforge/jobposition/entity/JobPosition.java
package com.careerforge.jobposition.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.careerforge.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 职位实体类
 * 存储JD分析信息，同一公司同一职位可被多个面试复用
 *
 * @author Azir
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName(value = "t_job_position", autoResultMap = true)
public class JobPosition extends BaseEntity {

    /**
     * 关联公司ID
     */
    private String companyId;

    /**
     * 职位名称
     */
    private String title;

    /**
     * JD原文
     */
    private String jdContent;

    /**
     * JD分析结果（JSON格式）
     */
    private String jdAnalysis;

    /**
     * 分析更新时间
     */
    private LocalDateTime jdAnalysisUpdatedAt;

    /**
     * 职位状态（用户手动设置）
     * draft/applied/interviewing/offered/rejected/withdrawn
     */
    private String status;

}
